bcl/src/bc_module_infra_grid.c

```diff
@@ -174,12 +175,41 @@ static void _bc_module_infra_grid_task_measure(void *param)
         {
             self->_state = BC_MODULE_INFRA_GRID_STATE_ERROR;
 
+            if (bc_tca9534a_init(&self->_tca9534, BC_I2C_I2C0 , 0x23))
+            {
+                self->_revision = BC_MODULE_INFRA_GRID_REVISION_R1_1;
+            }
+            else
+            {
+                self->_revision = BC_MODULE_INFRA_GRID_REVISION_R1_0;
+            }
+
+            if (self->_revision == BC_MODULE_INFRA_GRID_REVISION_R1_1)
+            {
+                bc_tca9534a_set_pin_direction(&self->_tca9534, BC_TCA9534A_PIN_P7, BC_TCA9534A_PIN_DIRECTION_OUTPUT);
+                bc_tca9534a_write_pin(&self->_tca9534, BC_TCA9534A_PIN_P7, 1);
+            }
+
+            // Update sleep flag
+            if (self->_enable_sleep != self->_cmd_sleep)
+            {
+                self->_enable_sleep = self->_cmd_sleep;
+            }
+
             if (self->_enable_sleep)
             {
-                // Sleep mode
-                if (!bc_i2c_memory_write_8b(self->_i2c_channel, self->_i2c_address, _BC_AMG88xx_PCLT, 0x10))
+                if (self->_revision == BC_MODULE_INFRA_GRID_REVISION_R1_0)
                 {
-                    goto start;
+                    // Sleep mode
+                    if (!bc_i2c_memory_write_8b(self->_i2c_channel, self->_i2c_address, _BC_AMG88xx_PCLT, 0x10))
+                    {
+                        goto start;
+                    }
+                }
+                else if (self->_revision == BC_MODULE_INFRA_GRID_REVISION_R1_1)
+                {
+                    // Revision 1.1 - disconnect power to infragrid
```

Comment:
Please, be cautious about names in comments - as it is a human text, it should be referred to as `Infra Grid Module`